hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestApacheClientConnectionPool.java

```diff
@@ -53,9 +66,27 @@ public void testSysPropAppliedPool() throws Exception {
     validatePoolSize(Integer.parseInt(customPoolSize));
   }
 
+  @Test
+  public void testPoolWithZeroSysProp() throws Exception {
+    final String customPoolSize = "0";
+    System.setProperty(HTTP_MAX_CONN_SYS_PROP, customPoolSize);
+    validatePoolSize(DEFAULT_HTTP_CLIENT_CONN_MAX_CACHED_CONNECTIONS);
+  }
+
+  @Test
+  public void testEmptySizePool() throws Exception {
+    Configuration configuration = new Configuration();
+    configuration.set(FS_AZURE_APACHE_HTTP_CLIENT_MAX_CACHE_CONNECTION_SIZE, "0");
+    AbfsConfiguration abfsConfiguration = new AbfsConfiguration(configuration, EMPTY_STRING);
+    try (KeepAliveCache keepAliveCache = new KeepAliveCache(abfsConfiguration)) {
+      Assertions.assertThat(keepAliveCache.put(Mockito.mock(HttpClientConnection.class))).isFalse();
```

Comment:
add .describedAs() statements here before the isFalse() and the isNull() below. minimal one like "cache.put()" and cache.get() are enough

Reply:
Make sense, have taken out all the assertions on cache.put and cache.get to the required methods. Description in assertion is added.